components/renku_data_services/namespace/orm.py

```diff
@@ -0,0 +1,143 @@
+"""SQLAlchemy's schemas for the group database."""
+
+from datetime import datetime
+from typing import Dict, Optional
+
+from sqlalchemy import DateTime, Integer, MetaData, String
+from sqlalchemy.orm import DeclarativeBase, Mapped, MappedAsDataclass, attribute_keyed_dict, mapped_column, relationship
+from sqlalchemy.schema import ForeignKey
+from ulid import ULID
+
+from renku_data_services.namespace import models
+from renku_data_services.users.orm import UserORM
+
+
+class BaseORM(MappedAsDataclass, DeclarativeBase):
+    """Base class for all ORM classes."""
+
+    metadata = MetaData(schema="common")
+
+
+class GroupORM(BaseORM):
+    """A Renku group."""
+
+    __tablename__ = "groups"
+
+    id: Mapped[str] = mapped_column("id", String(26), primary_key=True, default_factory=lambda: str(ULID()), init=False)
+    name: Mapped[str] = mapped_column("name", String(99), index=True)
+    created_by: Mapped[str] = mapped_column(ForeignKey(UserORM.keycloak_id), index=True, nullable=False)
+    creation_date: Mapped[datetime] = mapped_column("creation_date", DateTime(timezone=True))
+    ltst_ns_slug: Mapped["NamespaceORM"] = relationship(lazy="joined", join_depth=1)
+    ltst_ns_slug_id: Mapped[str] = mapped_column(ForeignKey("namespaces.id"), init=False, nullable=False)
```

Comment:
I don't think we should abbreviate as ltst here, it's uncommon enough that spelling it out increases readability. Especially since we have `latest` in other places, like `latest_slug` on `Namespace`